csharp/ql/src/Likely Bugs/ThreadUnsafeICryptoTransform.ql

```diff
@@ -0,0 +1,93 @@
+/**
+ * @name Class defines a field that uses an ICryptoTransform class in a way that would be unsafe for concurrent threads.
+ * @description The class has a field that directly or indirectly make use of a static System.Security.Cryptography.ICryptoTransform object.
+ *              Using this an instance of this class in concurrent threads is dangerous as it may not only result in an error, 
+ *              but under some circumstances may also result in incorrect results.
+ * @kind problem
+ * @problem.severity warning
+ * @precision medium
+ * @id cs/thread-unsafe-icryptotransform-field-in-class
+ * @tags concurrency
+ *       security
+ *       external/cwe/cwe-362
+ */
+
+import csharp
+
+class ICryptoTransform extends Class {
+  ICryptoTransform() {
+    this.getABaseType*().hasQualifiedName("System.Security.Cryptography", "ICryptoTransform")
+  }
+}
+
+predicate usesICryptoTransformType( Type t ) {
+  exists(  ICryptoTransform ict |
+    ict = t
+    or usesICryptoTransformType( t.getAChild*() )
```

Comment:
Since this predicate is recursive anyway, you can omit the `*`.

Reply:
Fixed